### ISSUE TITLE
Modify the code to enable the high-resolution training

### ISSUE THREAD

**Issue by KevinChen880723** [junyanz/pytorch-CycleGAN-and-pix2pix#1411]
Hello @junyanz, thanks for providing such a fantastic method! I studied your code and tried to write it from scratch. I found that the function "[backward_G](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/master/models/cycle_gan_model.py#L151)" can be modified to reduce RAM usage. [link](https://github.com/KevinChen880723/CycleGAN/blob/master/train.py#L100) is my modification; due to PyTorch accumulating the gradients by default, I calculate the loss for the two different generators separately. The Autograd package will release the computational graph for each loss when we execute `backward()` function. In this way, we will not store all of the computational graphs in the RAM simultaneously.

**Comment by cankur007** [comment]
@KevinChen880723 how about the results after training? Did you get good results in testing?

**Comment by KevinChen880723** [comment]
Hello @cankur007, sorry for the late reply.

By my experiments, the testing results of zebra2horse are similar to the official repo. You can check the visualizations in the following table.

<table>
  <tr>
    <td>Description</td>
    <td>Input image</td>
    <td>Result</td>
  </tr>
  <tr>
    <td>Horse to zebra</td>
    <td><img src="https://github.com/KevinChen880723/CycleGAN/blob/master/results/horse2zebra/horse.png?raw=true"></td>
    <td><img src="https://github.com/KevinChen880723/CycleGAN/blob/master/results/horse2zebra/horse2zebra.png?raw=true"></td>
  </tr>
  <tr>
    <td>Zebra to horse</td>
    <td><img src="https://github.com/KevinChen880723/CycleGAN/blob/master/results/horse2zebra/zebra.png?raw=true"></td>
    <td><img src="https://github.com/KevinChen880723/CycleGAN/blob/master/results/horse2zebra/zebra2horse.png?raw=true"></td>
  </tr>
</table>

Besides, I tried using CycleGAN to do the visual translation between the Cityscapes and the IDD dataset. You can also check the visualizations in the following table.

<table>
  <tr>
    <td>Domain</td>
    <td>Input image</td>
    <td>Output image</td>
  </tr>
  <tr>
    <td>Cityscapes to IDD</td>
    <td><img src="https://github.com/KevinChen880723/CycleGAN/blob/master/results/German2India/A2B/input.png?raw=true"></td>
    <td><img src="https://github.com/KevinChen880723/CycleGAN/blob/master/results/German2India/A2B/ResidualBlock15_withIdentity_Crop1024.png?raw=true"></td>
  </tr>
  <tr>
    <td>IDD to Cityscapes</td>
    <td><img src="https://github.com/KevinChen880723/CycleGAN/blob/master/results/German2India/B2A/original.png?raw=true"></td>
    <td><img src="https://github.com/KevinChen880723/CycleGAN/blob/master/results/German2India/B2A/Residual15_withIdentity_Crop1024.png?raw=true"></td>
  </tr>
</table>
